inc/functions.php

```diff
@@ -5,13 +5,46 @@
  * @package Core_Sitemaps
  */
 
+/**
+ * Retrieves the current Sitemaps server instance.
+ *
+ * @return Core_Sitemaps Core_Sitemaps instance.
+ */
+function core_sitemaps_get_server() {
+	/**
+	 * Global Core Sitemaps instance.
+	 *
+	 * @var Core_Sitemaps $core_sitemaps
+	 */
+	global $core_sitemaps;
+
+	// If there isn't a global instance, set and bootstrap the sitemaps system.
+	if ( empty( $core_sitemaps ) ) {
+		$core_sitemaps = new Core_Sitemaps();
+		$core_sitemaps->init();
+
+		/**
+		 * Fires when initializing the Core_Sitemaps object.
+		 *
+		 * Additional sitemaps should be registered on this hook.
+		 *
+		 * @since 0.1.0
+		 *
+		 * @param core_sitemaps $core_sitemaps Server object.
+		 */
+		do_action( 'core_sitemaps_init', $core_sitemaps );
```

Comment:
Just so it's clear, this is now the replacement for `core_sitemaps_register`?

Reply:
Correct. I think this action pattern is more clear and allows functionality to be extended/modified when the system has initialized, and can be used internally by the sitemaps system to better control the flow of actions that are initialized specifically by the sitemaps system.